addons/website_sale_collect/tests/test_sale_order.py

```diff
@@ -0,0 +1,54 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command
+from odoo.exceptions import ValidationError
+from odoo.tests import tagged
+
+from odoo.addons.website_sale_collect.tests.common import InStoreCommon
+
+
+@tagged('post_install', '-at_install')
+class TestSaleOrder(InStoreCommon):
+
+    def test_set_delivery_line_recomputes_so_wh_when_dm_is_not_in_store(self):
+        self.website.warehouse_id = self.warehouse_1
+        so = self._create_so_in_store_dm(warehouse_id=self.warehouse_2.id)
+        so.set_delivery_line(self.free_delivery, 0)
+        self.assertEqual(so.warehouse_id, self.warehouse_1)
+
+    def test_set_pickup_location_sets_wh_on_so(self):
+        so = self._create_so_in_store_dm()
+        so._set_pickup_location('{"id":' + str(self.warehouse_1.id) + '}')
+        self.assertEqual(so.warehouse_id, self.warehouse_1)
+
+    def test_get_unavailable_product_lines(self):
+        cart = self._create_so_in_store_dm(
+            order_line=[
+                Command.create({
+                    'product_id': self.product_A.id,
+                    'product_uom_qty': 5.0,
+                }),
+                Command.create({
+                    'product_id': self.product_B.id,
+                    'product_uom_qty': 5.0,
+                }),
+            ]
+        )
+        unavailable_ol = cart._get_unavailable_order_lines(self.warehouse_1.id)
+        self.assertIn(self.product_B.id, unavailable_ol.product_id.ids)
+        self.assertNotIn(self.product_A.id, unavailable_ol.product_id.ids)
+
+    def test_check_cart_is_ready_to_be_paid_raise_when_products_are_out_of_stock(self):
+        cart = self._create_so_in_store_dm(order_line=[Command.create({
+            'product_id': self.product_B.id,
+            'product_uom_qty': 5.0,
+        })])
+        cart.warehouse_id = self.warehouse_1
+        with self.assertRaises(ValidationError):
+            cart._check_cart_is_ready_to_be_paid()
+
+    def test_verify_updated_quantity_skipped_when_in_store_dm_is_activated(self):
```

Comment:
Well, thanks to this test, I took a look to your override of `_verify_updated_quantity` and I find it quite strange to bypass the check without verifying that pickup in store was chosen by the user :thinking:.

Reply:
Yeah, but where do you check later on that it's really the in store delivery that was chosen ?

Reply:
it is checked when clicked on 'pay' but it should be checked in the /checkout and there is a corresponding task https://www.odoo.com/odoo/4106/tasks/4328322

Reply:
Perfect then, mostly wanted to be sure we added some safety checks to avoid issues :+1: